sqlglot/parser.py

```diff
@@ -4463,6 +4463,14 @@ def _parse_qualify(self) -> t.Optional[exp.Qualify]:
             return None
         return self.expression(exp.Qualify, this=self._parse_assignment())
 
+    def _parse_connect_with_prior(self) -> t.Optional[exp.Expression]:
```

Comment:
I think we should probably simplify this. To my knowledge, only Snowflake and Oracle support the `CONNECT BY` clause, and while Snowflake allows `PRIOR` to be used as an identifier in other contexts, parsing it in the context of said clause should be relatively simple given their [syntax spec](https://docs.snowflake.com/en/sql-reference/constructs/connect-by#syntax).

```
    ...
    CONNECT BY [ PRIOR ] <col1_identifier> = [ PRIOR ] <col2_identifier>
           [ , [ PRIOR ] <col3_identifier> = [ PRIOR ] <col4_identifier> ]
           ...
```

I think we could probably just roll a customer parser for what comes the `CONNECT BY` to capture this without mutating the parser's state, which is a smell.

Reply:
Discussed in Slack– Snowflake supports other operators besides `=` and can also chain these using boolean connectors, so manually parsing them would end up being a pain.